handler/handler.go

```diff
@@ -141,12 +151,11 @@ func (h *ReindexRequestedHandler) getAndSendZebedeeDocsURL(ctx context.Context,
 func (h *ReindexRequestedHandler) getAndSendDatasetURLs(ctx context.Context, cfg *config.Config, datasetAPICli clients.DatasetAPIClient, reindexReqEvent *models.ReindexRequested, task chan taskDetails) {
 	log.Info(ctx, "extract and send dataset urls")
 	var wgDataset sync.WaitGroup
-	wgDataset.Add(4)
+	wgDataset.Add(3)
```

Comment:
The urlcount=… function is ranging over the datasetURLChan anyway, so it blocks until that is closed so in reality there's no need for this wg at all. It's not doing any harm, but will already be complete by the time the `sendExtractedDatasetURLs(…)` function is complete. You can safely remove it and all function params of the functions that use it if you like.